cyctl/cmd/serve.go

```diff
@@ -0,0 +1,113 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"github.com/spf13/cobra"
+	"log"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	"k8s.io/client-go/util/homedir"
+)
+
+var serveCMD = &cobra.Command{
+	Use:   "serve",
+	Short: "Start the Cyclops UI",
+	Long:  "Start the Cyclops UI by forwarding the Cyclops UI service's port to a local port",
+	Run: func(cmd *cobra.Command, args []string) {
+		startPortForwarding()
+	},
+}
+
+func startPortForwarding() {
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	service, err := clientset.CoreV1().Services("cyclops").Get(context.TODO(), "cyclops-ui", metav1.GetOptions{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	podList, err := clientset.CoreV1().Pods("cyclops").List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labels.Set(service.Spec.Selector).String(),
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	if len(podList.Items) == 0 {
+		log.Fatal("no pods found for service cyclops-ui in namespace cyclops")
+	}
+
+	rand.Seed(time.Now().UnixNano())
+	podIndex := rand.Intn(len(podList.Items))
+	podName := podList.Items[podIndex].Name
+
+	roundTripper, upgrader, err := spdy.RoundTripperFor(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	path := fmt.Sprintf("/api/v1/namespaces/%s/pods/%s/portforward", "cyclops", podName)
+	hostIP := strings.TrimLeft(config.Host, "htps:/")
```

Comment:
Why are you trimming `htps:/`?

Reply:
Because it already adds `https` when I try to establish the connection:

![image](https://github.com/cyclops-ui/cyclops/assets/95130644/e3aa89fd-7487-49b0-bd36-842b81030f97)

Reply:
Is it added because of this line?
```
url.URL{Scheme: "https", Path: path, Host: hostIP}
```

You are trimming `htps:/` which misses a `t`

Reply:
It handles the duplicate values